internals/maintenance_scripts.py

```diff
@@ -1152,3 +1152,53 @@ def get_template_data(self, **kwargs) -> str:
       f.outstanding_notifications = 0
     ndb.put_multi(notified_features)
     return f'{len(notified_features)} reverted to 0 outstanding notifications.'
+
+
+class ResetStaleShippingMilestones(FlaskHandler):
+  """Reset the shipping milestones of features have not been verified after 4+ notifications."""
+
+  def _reset_milestone(self, stage: Stage, field: str, activity: Activity):
+    old_value = getattr(stage.milestones, field) if stage.milestones else None
+    if old_value and old_value > 140:
+      old_value = getattr(stage.milestones, field)
+      # Capture the old value in an Amendment.
+      activity.amendments.append(
+        Amendment(field_name=field, old_value=str(old_value), new_value=None)
+      )
+      setattr(stage.milestones, field, None)
+
+  def get_template_data(self, **kwargs) -> str:
+    self.require_cron_header()
+
+    num_features_reset = 0
+    stale_features: list[FeatureEntry] = FeatureEntry.query(
+      FeatureEntry.outstanding_notifications >= 4
+    ).fetch()
+    entities_to_update: list[ndb.Model] = []
+    for f in stale_features:
+
+      # Get all the shipping stages of the stale feature and reset any
+      # milestones that are set.
+      stages: list[Stage] = Stage.query(
+        Stage.feature_id == f.key.integer_id(),
+        Stage.stage_type == STAGE_TYPES_SHIPPING[f.feature_type]
+      ).fetch()
+      for s in stages:
+        # Create an activity that shows all the shipping milestones have been set to null.
+        activity = Activity(
+          feature_id=f.key.integer_id(),
+          content='Shipping milestones were unset due to failure to verify accuracy.')
+        if s.milestones:
+          self._reset_milestone(s, 'desktop_first', activity)
+          self._reset_milestone(s, 'android_first', activity)
+          self._reset_milestone(s, 'ios_first', activity)
+          self._reset_milestone(s, 'webview_first', activity)
+          entities_to_update.append(s)
+          entities_to_update.append(activity)
```

Comment:
I think you need to only add the Activity if it has some Amendments.  Otherwise, when time this Cron job runs, a feature with milestones before 140 will get an empty Activity.